docs/guide/database-queries.md

```diff
@@ -0,0 +1,273 @@
+
+The `toChoices` query method builds on `toDistinct` by returning each choice as an object with `label` and `value` properties. This can be useful when populating a select or other input field with options for a doc type property. For example, this is used for the document manager modal filter UI in Apostrophe.
+
+`toChoices` accepts an options object as an optional second argument. Setting `counts: true` in that options object will include a `count` property on each returned choice indicating how many documents match that choice.
+
+```javascript
+const teamOffices = await self.find(req, { type: 'team' })
+  .toDistinct('office', { count: true });
+```
+
+## Query across modules
+
+The `find()` method in doc type modules is easy to use within each module as `self.find`. However this assumes we're looking for only content that is governed by the module where the method is called. There are two main ways to write these database queries from one module and get content managed by a separate module.
+
+### Using another module's `find` method
+
+The `self` object available in any Apostrophe module's customization functions can access other module managers on the `self.apos.modules` object. For example, if I'm working in the `article` module and want to query `author` pieces, I can access the `author` module with this:
+
+```javascript
+self.apos.modules.author
+```
+
+`self.apos.modules.author` is essentially the same as `self` would be if we were operating within the `author` module. Therefore we can query author pieces directly by using that module's `find` method.
+
+```javascript
+const activeAuthors = self.apos.modules.author.find(req, { active: true })
+  .toArray();
+```
+
+If a doc type module has been [assigned an alias](/reference/module-api/module-options.md#alias), the doc type's manager will be directly on `self.apos`. For example, the `@apostrophecms/page` module is available as `self.apos.page` because `'page'` is its alias option value.
+
+### Querying multiple doc types
+
+As mentioned earlier, doc type modules' `find` methods will automatically restrict results to that document type. Sometimes we may want to fetch documents that match a query regardless of what doc type they are (e.g., all content tagged with a certain term). Or we may use a querying function across contexts and we don't know what doc type we will be looking for.
+
+There is a module that governs all content documents: `@apostrophecms/doc`, which is aliased as `'doc'`. Its `find` method works the same as any individual doc type's, but it is not restricted to any one document type. We can then use it to look for a particular type with the `type` property or pass it criteria and get documents of multiple types in the results.
+
+```javascript
+const featuredContent = self.apos.doc.find(req, { featured: true })
+  .toArray();
+
+
+const featuredByType = self.apos.doc.find(req, {
+  featured: true,
+  type: selectedType // A hypothetical option a user selected.
```

Comment:
We should mention the limitations here. Pieces won't get their `_url` property this way, so you might have to make an additional query with each type-specific `find` after doing your blanket query across types, perhaps to get them all sorted by search match quality... which is really the job of the search module, and almost but not quite the only use case for this that most developers will encounter.